.github/goodtables_report.py

```diff
@@ -62,30 +68,32 @@ def is_invalid_schema():
                     "fallback": f'```{str(err)}```',
                     "color": "danger",
                     "fields":[{
-                       "title": os.environ['SCHEMA_PATH'],
+                       "title": SCHEMA_PATH,
                        "value": f'```{str(err)}```'
                     }]
                 }
             ]
         }
 
         # post stringified payload to url
-        requests.post(os.environ['SLACK_GOODTABLES_WEBHOOK'], json.dumps(j_payload))
-        if os.environ.get('SLACK_GOODTABLES_WEBHOOK_2'):
-            requests.post(os.environ['SLACK_GOODTABLES_WEBHOOK_2'], json.dumps(j_payload))
+        requests.post(WEBHOOK, json.dumps(j_payload))
+        if WEBHOOK_2:
+            requests.post(WEBHOOK_2, json.dumps(j_payload))
 
         return True
     
     return False
 
 
 def main():
-    if is_invalid_schema():
-        return
-
-    with open(ERROR_FILE) as f: 
+    # Load validation file
+    with open(VALIDATION_OUTPUT_FILE) as f: 
         error_json = json.load(f)
 
+    # Check if schema is valid
+    if is_invalid_schema(error_json):
+        return
+
```

Comment:
Took me a minute reading this to see how you changed it (I think because it wasn't quite what I expected). It doesn't seem logical to me for the invalid schema check to depend on or even know about the validation output (the schema is valid or not independent of that; it's just that we only care about it in one of those cases). 

You have the basic logic right  — just move the error json table check out of the method and flip it so you only call the `is_invalid_schema`  check when they aren't present.

Reply:
Great, I'll move it!